gocourse04/SmartZoo/main.go

```diff
@@ -0,0 +1,255 @@
+/*
+При розробці системи «Розумний зоопарк» техлід ще не вирішив, яку базу даних використовувати і для реалізації
+прототипа вирішили зберігати дані в памʼяті програми, а саме — в мапах. Треба побудувати складну структуру мап,
+відобразивши деяку ієрархію частин зоопарку. Наприклад, зоопарк ділиться на території: копитні, пернаті,
+примати… та інше. Кожна територія має декілька загонів для кожного виду тварин. Кожен загін може мати ділянку
+для перебування тварини і технічне приміщення, де можуть зберігатися приладдя для роботи з тваринами. На кожній
+території є певна кількість тварин, а технічне приміщення «здатне» виконувати деякі функції, такі як прибирання і
+годування тварин. Реалізувати функції пошуку тварини за імʼям або ID, переміщення тварини із загону в загін,
+годування тварини.
+*/
+package main
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	areaClean   = "tech room"
+	sectorClean = 6
+)
+
+type Areas map[string]Area
+
+// Територія по  типу (копитні, пернаті, примати)
+type Area struct {
+	Name    string
+	Sectors map[string]Sector
+}
+
+// Загон для виду тварин. У кожному загоні є технічне приміщення (наприклад годування)
+type Sector struct {
+	ID      int
+	Animals []Animal
+}
+
+type Animal struct {
+	ID   int
+	Name string
+}
+
+type Zoo struct {
+	Areas Areas
+}
+
+func NewZoo(areasData map[string]Area) *Zoo {
```

Comment:
Можна просто `areas`, так коротко та зрозуміло:
```suggestion
func NewZoo(areas map[string]Area) *Zoo {
```